### PR TITLE
deps: bump reth to v0.1.0.alpha-10

### PR DESCRIPTION
**Pull request by chirag-bgh**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,20 +10,20 @@ version = "0.3.0"
...
 beacon-api-client = { git = "https://github.com/ralexstokes/ethereum-consensus", rev = "afbf74216b49d30562aa3e966b9daa9aa416d534" }
 
-reth-payload-builder = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-primitives = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-provider = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-rpc-types = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-transaction-pool = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-interfaces = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-reth-revm = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
+reth-payload-builder = { git = "https://github.com/paradigmxyz/reth", rev = "1b16d804ef01f4ec3c25e7986381c22739c105b9" }
```
Comment (ralexstokes) [inline Cargo.toml:13]: can we just say `v0.1.0-alpha10` here instead of the hash?

File: Cargo.toml [hunk 2]
```diff
@@ -10,20 +10,20 @@ version = "0.3.0"
...
-reth = { git = "https://github.com/paradigmxyz/reth", rev = "900ada5aaa4b5d4a633df78764e7dd7169a13405" }
-revm = { git = "https://github.com/bluealloy/revm/", branch = "release/v25" }
+reth = { git = "https://github.com/paradigmxyz/reth", rev = "1b16d804ef01f4ec3c25e7986381c22739c105b9" }
+revm = { git = "https://github.com/bluealloy/revm/", rev = "516f62cc" }
 
 [patch.crates-io]
-revm = { git = "https://github.com/bluealloy/revm/", branch = "release/v25" }
-revm-primitives = { git = "https://github.com/bluealloy/revm/", branch = "release/v25" }
+revm = { git = "https://github.com/bluealloy/revm/", rev = "516f62cc" }
+revm-primitives = { git = "https://github.com/bluealloy/revm/", rev = "516f62cc" }
```
Comment (ralexstokes) [inline Cargo.toml:26]: did this get cleaned up in reth? at the time of alpha8, we needed the v25 branch but we may be able to just inherit the default now

this? https://github.com/paradigmxyz/reth/blob/main/Cargo.toml#L110C15-L110C15

File: mev-build-rs/src/reth_builder/error.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ pub enum Error {
     #[error(transparent)]
     Reth(#[from] RethError),
     #[error("evm execution error: {0:?}")]
-    Execution(EVMError<reth_interfaces::Error>),
+    Execution(EVMError<reth_interfaces::RethError>),
```
Comment (ralexstokes) [inline mev-build-rs/src/reth_builder/error.rs:31]:
```suggestion
    Execution(EVMError<RethError>),
```

**Review by ralexstokes — COMMENTED** [review]
this looks great!

have some questions about the deps

File: Cargo.toml (hunk 2 above)
Comment (chirag-bgh) [reply Cargo.toml:26]: There were some issues with latest revm version so used the revm version in alpha 10.

Comment (ralexstokes) [reply Cargo.toml:26]: I see, this works; let's target a tagged release for now!

**Pull request merged**